catalogue/webapp/components/ConfirmItemRequest/ConfirmItemRequest.tsx

```diff
@@ -184,37 +202,91 @@ const ErrorDialog: FunctionComponent<ErrorDialogProps> = ({ setIsActive }) => (
   </>
 );
 
+type RequestingState = null | 'requesting' | 'confirmed' | 'error';
+
 const ConfirmItemRequest: FunctionComponent<Props> = props => {
   const openButtonRef = useRef<HTMLButtonElement>(null);
-  const { item, work, setIsActive, ...modalProps } = props;
-  const [isConfirmed, setIsConfirmed] = useState(false);
-  const [isLoading, setIsLoading] = useState(false);
-  const [isError, setIsError] = useState(false); // TODO: implement this if something goes wrong with API call
-
+  const { item, work, setIsActive, user, initialHoldNumber, ...modalProps } =
+    props;
+  const [requestingState, setRequestingState] = useState<RequestingState>(null);
+  const [currentHoldNumber, setCurrentHoldNumber] = useState(initialHoldNumber);
   function innerSetIsActive(value: boolean) {
     if (value) {
       setIsActive(true);
-    } else if (isLoading) {
+    } else if (requestingState === 'requesting') {
       // disable close dialog button during api call
     } else {
       setIsActive(false);
-      setIsConfirmed(false);
-      setIsError(false);
     }
   }
 
-  function confirmRequest() {
-    setIsLoading(true);
-    // setIsError(true); // uncomment to test the error UI
+  useEffect(() => {
+    setCurrentHoldNumber(initialHoldNumber);
+  }, [initialHoldNumber]);
```

Comment:
Does this ever change?

Reply:
yes, when the PhysicalItemDetails component renders and the userHolds are updated

Reply:
Worth noting (sorry for this broken record, but if we come back to it or someone else works on this, it would be helpful to know as it's not immediately obvious).